### PR TITLE
Remove docker files

### PR DESCRIPTION
**Pull request by feedmeapples**
**What changed?**

Removes docker files, moved to https://github.com/temporalio/docker-builds

**Why?**

`docker-builds` handles build details of tctl and temporal repos

**How did you test it?**

ran make scripts manually

**Potential risks**

cicd pipelines failure

**Is hotfix candidate?**

### CODE REVIEW

File: develop/buildkite/pipeline.yml [hunk 1]
```diff
@@ -11,12 +11,6 @@ steps:
 
   - wait
 
-  - label: ":docker: build docker images"
-    agents:
-      queue: "default"
-      docker: "*"
-    command: "docker build ."
-
```
Comment (alexshtin) [inline develop/buildkite/pipeline.yml:19]: Yes, we can't build docker here any more but you also shouldn't remove it. Let's change this to:
```suggestion
  - label: ":docker: build without cgo"
    agents:
      queue: "default"
      docker: "*"
    command: "CGO_ENABLED=0 make bins"
    plugins:
      - docker-compose#v3.8.0:
          run: build
          config: ./develop/buildkite/docker-compose.yml    
```

File: docker/README.md [hunk 2]
```diff
@@ -12,11 +12,7 @@ To run docker image with dependencies:
 
 ## Build docker image for any commit
 
-Replace **YOUR_TAG** and **YOUR_CHECKOUT_COMMIT** in the below command to build:
-```bash
-git checkout YOUR_CHECKOUT_COMMIT
-docker build . -t temporalio/auto-setup:YOUR_TAG --target temporal-auto-setup
-```
+Refer to [docker-builds](https://github.com/temporalio/docker-builds) to build the images manually
```
Comment (alexshtin) [inline docker/README.md:15]:
```suggestion
Refer to [docker-builds](https://github.com/temporalio/docker-builds) to build the images manually.
```

File: docker/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Build docker image
```
Comment (alexshtin) [inline docker/README.md:1]: `docker` dir should go away. You can leave `config_templage.yaml` for now (I will move it later) but the content of `README.md` should be migrated to the root one (with proper modifications, of course). Or to some `DOCKER.md` if you don't want to pollute main file.

File: Makefile [hunk 4]
```diff
@@ -457,42 +453,6 @@ fossa-test:
 
 build-fossa: bins fossa-analyze fossa-delay fossa-test
 
-##### Docker #####
-docker-server:
```
Comment (alexshtin) [inline Makefile:461]: I don't see this block in the new repo. It is convenient to build images locally.

Comment (feedmeapples) [reply Makefile:461]: added https://github.com/temporalio/docker-builds/pull/16

File: docker/README.md (hunk 3 above)
Comment (feedmeapples) [reply docker/README.md]: moved docker readme here https://github.com/temporalio/docker-builds/pull/17
added comment about docker images https://github.com/temporalio/temporal/pull/2456/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R32

File: develop/buildkite/pipeline.yml [hunk 5]
```diff
@@ -11,11 +11,15 @@ steps:
 
   - wait
 
-  - label: ":docker: build docker images"
+  - label: ":docker: build without cgo"
```
Comment (alexshtin) [inline develop/buildkite/pipeline.yml:14]:
```suggestion
  - label: ":golang: build without cgo"
```

File: README.md [hunk 6]
```diff
@@ -29,6 +29,8 @@ docker-compose up
 
 Refer to Temporal [docker-compose](https://github.com/temporalio/docker-compose) repo for more advanced options.
 
+For more details on Docker images refer to [docker-builds](https://github.com/temporalio/docker-builds)
```
Comment (alexshtin) [inline README.md:32]:
```suggestion
For more details on Docker images refer to [docker-builds](https://github.com/temporalio/docker-builds) repo.
```

**Review by alexshtin — APPROVED** [review]

**Pull request merged**